addons/mrp/models/mrp_bom.py

```diff
@@ -92,6 +94,49 @@ def _compute_possible_product_template_attribute_value_ids(self):
         for bom in self:
             bom.possible_product_template_attribute_value_ids = bom.product_tmpl_id.valid_product_template_attribute_line_ids._without_no_variant_attributes().product_template_value_ids._only_active()
 
+    @api.depends('bom_line_ids.product_id', 'bom_line_ids.product_qty', 'bom_line_ids.product_uom_id')
+    def _compute_quantities(self):
+        res = self._compute_quantities_dict()
+        for bom in self:
+            bom.qty_available = res[bom.id]['qty_available']
+            bom.virtual_available = res[bom.id]['virtual_available']
+
+    def _compute_quantities_dict(self):
+        res = {}
+        for bom in self:
+            _boms, bom_sub_lines = bom.explode(bom.product_id or bom.product_tmpl_id.product_variant_id, 1)
```

Comment:
allow to pass in the context witch variante (if you have a template with variante)

```python
product = self.env['product.product'].browse(self.env.context.get('product_id')) or bom.product_id or bom.product_tmpl_id.product_variant_id
_boms, bom_sub_lines = bom.explode(product)
```